From 76f2207399d09b198aafceef3e6a3dbdd460c1eb Mon Sep 17 00:00:00 2001
From: Valentine Barshak <valentine.barshak@cogentembedded.com>
Date: Mon, 17 Dec 2018 16:47:51 +0300
Subject: [PATCH] arm64: dts: renesas: ulcb-kf: Move panel configuration to
 ulcb-kf-panel.dtsi

This moves LVDS panel configuration from ulcb-kf.dtsi
to ulcb-kf-panel.dtsi. This various LVDS panel setup
a bit easier.

Signed-off-by: Valentine Barshak <valentine.barshak@cogentembedded.com>
---
 arch/arm64/boot/dts/renesas/ulcb-kf-panel.dtsi | 41 ++++++++++++++++++++++++++
 arch/arm64/boot/dts/renesas/ulcb-kf.dtsi       | 34 +++------------------
 2 files changed, 45 insertions(+), 30 deletions(-)
 create mode 100644 arch/arm64/boot/dts/renesas/ulcb-kf-panel.dtsi

diff --git a/arch/arm64/boot/dts/renesas/ulcb-kf-panel.dtsi b/arch/arm64/boot/dts/renesas/ulcb-kf-panel.dtsi
new file mode 100644
index 0000000..1bcb246
--- /dev/null
+++ b/arch/arm64/boot/dts/renesas/ulcb-kf-panel.dtsi
@@ -0,0 +1,41 @@
+/*
+ * Device Tree Source for the Kingfisher (ULCB extension) LVDS panel
+ *
+ * Copyright (C) 2018 Renesas Electronics Corp.
+ * Copyright (C) 2018 Cogent Embedded, Inc.
+ *
+ * This file is licensed under the terms of the GNU General Public License
+ * version 2.  This program is licensed "as is" without any warranty of any
+ * kind, whether express or implied.
+ */
+
+&lvds {
+	compatible = "panel-lvds";
+
+	width-mm = <210>;
+	height-mm = <158>;
+
+	data-mapping = "jeida-24";
+
+	panel-timing {
+		/* 1280x800 @60Hz Boundary Devices BD101LCC1 compatible panel */
+		clock-frequency = <65000000>;
+		hactive = <1280>;
+		vactive = <800>;
+		hsync-len = <40>;
+		hfront-porch = <80>;
+		hback-porch = <40>;
+		vfront-porch = <14>;
+		vback-porch = <14>;
+		vsync-len = <4>;
+	};
+};
+
+&gpio2 {
+	bl_pwm {
+		gpio-hog;
+		gpios = <3 GPIO_ACTIVE_HIGH>;
+		output-high;
+		line-name = "BL PWM 100%";
+	};
+};
diff --git a/arch/arm64/boot/dts/renesas/ulcb-kf.dtsi b/arch/arm64/boot/dts/renesas/ulcb-kf.dtsi
index 0aa9009..aaa6ba8 100644
--- a/arch/arm64/boot/dts/renesas/ulcb-kf.dtsi
+++ b/arch/arm64/boot/dts/renesas/ulcb-kf.dtsi
@@ -211,27 +211,7 @@
 		};
 	};
 
-	lvds {
-		compatible = "panel-lvds";
-
-		width-mm = <210>;
-		height-mm = <158>;
-
-		data-mapping = "jeida-24";
-
-		panel-timing {
-			/* 1280x800 @60Hz */
-			clock-frequency = <65000000>;
-			hactive = <1280>;
-			vactive = <800>;
-			hsync-len = <40>;
-			hfront-porch = <80>;
-			hback-porch = <40>;
-			vfront-porch = <14>;
-			vback-porch = <14>;
-			vsync-len = <4>;
-		};
-
+	lvds: lvds {
 		port {
 			lvds_in: endpoint {
 				remote-endpoint = <&lvds0_out>;
@@ -364,15 +344,6 @@
 	status = "okay";
 };
 
-&gpio2 {
-	bl_pwm {
-		gpio-hog;
-		gpios = <3 GPIO_ACTIVE_HIGH>;
-		output-high;
-		line-name = "BL PWM 100%";
-	};
-};
-
 &gpio4 {
 	most_rst {
 		gpio-hog;
@@ -1445,5 +1416,8 @@
 	status = "okay";
 };
 
+/* set LVDS panel timings and pins */
+#include "ulcb-kf-panel.dtsi"
+
 
 
-- 
2.7.4

